src/javascript/static/pages/home.js

```diff
@@ -60,9 +62,21 @@ const Home = (() => {
             $('#signup_error').setVisibility(1).text(error.message);
         } else if (isBinaryApp()) {
             BinaryPjax.load(urlFor('new_account/virtualws'));
+            GTM.pushDataLayer({
+                event                   : 'email_submit',
+                email_submit_input      : response.echo_req.verify_email,
+                email_submit_days_passed: moment().diff(moment(localStorage.getItem('date_first_contact')), 'days'),
+                email_submit_source     : 'desktop app',
+            });
         } else {
             $('.signup-box div').replaceWith($('<p/>', { text: localize('Thank you for signing up! Please check your email to complete the registration process.'), class: 'gr-10 gr-centered center-text' }));
             $('#social-signup').setVisibility(0);
+            GTM.pushDataLayer({
+                event                   : 'email_submit',
+                email_submit_input      : response.echo_req.verify_email,
+                email_submit_days_passed: moment().diff(moment(localStorage.getItem('date_first_contact')), 'days'),
+                email_submit_source     : 'binary.com',
+            });
```

Comment:
why not just declare the common variable and push the email submit source to it later then send it once?

Reply:
if we never get an error then sure 👍

Reply:
it can be in a bigger else wrapper